test/scheduler/tainttoleration_integration_test.go

```diff
@@ -0,0 +1,702 @@
+		tolerateClusters := []string{memberCluster1EastProd, memberCluster2EastProd}
+		// The scheduler is designed to produce only deterministic decisions; if there are no
+		// comparable scores available for selected clusters, the scheduler will rank the clusters
+		// by their names.
+		wantFilteredClusters := []string{memberCluster1EastProd, memberCluster2EastProd, memberCluster3EastCanary, memberCluster4CentralProd, memberCluster5CentralProd, memberCluster6WestProd, memberCluster7WestCanary, memberCluster8UnhealthyEastProd, memberCluster9LeftCentralProd}
+		wantPickedClustersAfter := []string{memberCluster1EastProd, memberCluster2EastProd}
+		wantFilteredClustersAfter := []string{memberCluster3EastCanary, memberCluster4CentralProd, memberCluster5CentralProd, memberCluster6WestProd, memberCluster7WestCanary, memberCluster8UnhealthyEastProd, memberCluster9LeftCentralProd}
+
+		BeforeAll(func() {
+			// Ensure that no bindings have been created so far.
+			noBindingsCreatedActual := noBindingsCreatedForCRPActual(crpName)
+			Consistently(noBindingsCreatedActual, consistentlyDuration, consistentlyInterval).Should(Succeed(), "Some bindings have been created unexpectedly")
+
+			// Add taints to some member clusters 1, 2.
+			addTaintsToMemberClusters(taintClusters, buildTaints(taintClusters))
+
+			affinity := &placementv1beta1.Affinity{
+				ClusterAffinity: &placementv1beta1.ClusterAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &placementv1beta1.ClusterSelector{
+						ClusterSelectorTerms: []placementv1beta1.ClusterSelectorTerm{
+							{
+								LabelSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										regionLabel: "east",
+									},
+									MatchExpressions: []metav1.LabelSelectorRequirement{
+										{
+											Key:      envLabel,
+											Operator: metav1.LabelSelectorOpIn,
+											Values: []string{
+												"prod",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+
+			// Create a CRP of the PickN placement type, along with its associated policy snapshot, no tolerations specified.
+			createPickNCRPWithPolicySnapshot(crpName, numOfClusters, affinity, nil, policySnapshotName1, nil)
+		})
+
+		It("should add scheduler cleanup finalizer to the CRP", func() {
+			finalizerAddedActual := crpSchedulerFinalizerAddedActual(crpName)
+			Eventually(finalizerAddedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to add scheduler cleanup finalizer to CRP")
+		})
+
+		It("should create N bindings", func() {
+			hasNScheduledOrBoundBindingsActual := hasNScheduledOrBoundBindingsPresentActual(crpName, 0)
+			Eventually(hasNScheduledOrBoundBindingsActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to create N bindings")
+			Consistently(hasNScheduledOrBoundBindingsActual, consistentlyDuration, consistentlyInterval).Should(Succeed(), "Failed to create N bindings")
```

Comment:
not need to check the eventually, as you've already checked the consistently. duplicate functionality

Reply:
my bad, we need the eventually here to make sure the binding has been created, and then use the consistently to make sure it won't be updated during the "duration". Eventually is required.